demo/api/src/products/custom-product.resolver.ts

```diff
@@ -10,18 +11,18 @@ import { Product, ProductStatus } from "./entities/product.entity";
 export class CustomProductResolver {
     constructor(
         @InjectRepository(Product) private readonly repository: EntityRepository<Product>,
-        private readonly mailerService: MailerService,
+        private readonly mailTemplateService: MailTemplateService,
+        private readonly productPublishedMail: ProductPublishedMail,
     ) {}
 
     @Mutation(() => Boolean)
     async publishAllProducts(): Promise<boolean> {
+        const countProductPublished = await this.repository.count({ status: { $ne: ProductStatus.Published } });
         await this.repository.nativeUpdate({ status: { $ne: ProductStatus.Published } }, { status: ProductStatus.Published });
 
-        await this.mailerService.sendMail({
-            mailTypeForLogging: "products-published",
-            to: "product-manager@comet-dxp.com",
-            cc: "vice-product-manager@comet-dxp.com",
-            subject: "All products have been published",
+        await this.mailTemplateService.sendMail(this.productPublishedMail, {
+            recipient: { name: "Product Manager", email: "product-manager@comet-dxp.com" },
+            countProductPublished: countProductPublished,
         });
```

Comment:
alternative could be:
```suggestion
        await this.mailerService.sendMail({
            mailTypeForLogging: "foo",
            ....this.productPublishedMail.generateMail({
                recipient: { name: "Product Manager", email: "product-manager@comet-dxp.com" },
                countProductPublished: countProductPublished,
            }
        }));
```

One service less devs need to know... don't know if it is better....